### ISSUE TITLE
[0.13.10] unexpected java.lang.NoSuchFieldError when use nested case class

### ISSUE THREAD

**Issue by xuwei-k** [sbt/sbt#2411]
`project/build.properties`

```
sbt.version=0.13.10-RC1
```

`A.scala`

``` scala
package example

class A {
  case class B(x: Int)

  def c = B
}

object A {
  def main(args: Array[String]): Unit = {
    (new A).c
  }
}
```

result

```
[error] (run-main-15) java.lang.NoSuchFieldError: B$module
java.lang.NoSuchFieldError: B$module
        at example.A.B(A.scala:4)
        at example.A.c(A.scala:6)
        at example.A$.main(A.scala:11)
        at example.A.main(A.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
[trace] Stack trace suppressed: run last compile:run for the full output.
java.lang.RuntimeException: Nonzero exit code: 1
        at scala.sys.package$.error(package.scala:27)
[trace] Stack trace suppressed: run last compile:run for the full output.
[error] (compile:run) Nonzero exit code: 1
```

There is same problem in playframework, scalikejdbc, skinny-framework, cats.
- https://travis-ci.org/xuwei-k/Play20/jobs/104261656#L497
- https://travis-ci.org/xuwei-k/scalikejdbc/jobs/104293183#L1193
- https://travis-ci.org/seratch/skinny-framework/jobs/104275385#L2107
- https://travis-ci.org/xuwei-k/cats/jobs/104294044#L2702

**Comment by eed3si9n** [comment]
I should check if the actual `*.class` file is there after `compile`.

cc @adriaanm, @gkossakowski, @Duhemm

**Comment by dwijnand** [comment]
Using https://github.com/retronym/libscala/blob/master/git-java/textconv-javap the bytecode diff when switching from sbt 0.13.9 to 0.13.10-RC1 is:

``` diff
 target/scala-2.10/classes/example/A.class | Bin 2287 -> 2279 bytes
 1 file changed, 0 insertions(+), 0 deletions(-)

Warning: Binary file qFNMdb_A contains example.A
Warning: Binary file qFNMdb_A contains example.A
Warning: Binary file A contains example.A
Warning: Binary file A contains example.A
diff --git a/target/scala-2.10/classes/example/A.class b/target/scala-2.10/classes/example/A.class
index c4da4cd..5c6aaa9 100644
--- a/target/scala-2.10/classes/example/A.class
+++ b/target/scala-2.10/classes/example/A.class
@@ -1,9 +1,6 @@

 Compiled from "A.scala"
 public class example.A {
-  private volatile example.A$B$ B$module;
-    descriptor: Lexample/A$B$;
-
   public static void main(java.lang.String[]);
     descriptor: ([Ljava/lang/String;)V
     Code:
```

**Comment by dwijnand** [comment]
Note that this is using the default `scalaVersion` for sbt 0.13.10-RC1: Scala 2.10.6

**Comment by eed3si9n** [comment]
I think I'm seeing this at 5934350 using scripted.

```
$ cat sbt/src/sbt-test/source-dependencies/nested-case-class/A.scala
package example

class A {
  case class B(x: Int)
  def c = B
}
object A {
  def main(args: Array[String]): Unit = {
    (new A).c
  }
}
$ cat sbt/src/sbt-test/source-dependencies/nested-case-class/test
> run
```

/cc @Duhemm

**Comment by eed3si9n** [comment]
I'm seeing it at 46058029b5ec0c809f29f77dcb59e8a25166d69f but not the commit before it.

**Comment by Duhemm** [comment]
Fixed in #2418
